test/jdk/com/sun/management/OperatingSystemMXBean/TestTotalSwap.java

```diff
@@ -71,19 +57,31 @@ public class TestTotalSwap {
     private static final long MAX_SIZE_FOR_PASS = Long.MAX_VALUE;
 
     public static void main(String args[]) throws Throwable {
+
         // yocto might ignore the request to report swap size in bytes
         boolean swapInKB = mbean.getVersion().contains("yocto");
 
-        long expected_swap_size = getSwapSizeFromOs();
-
         long min_size = mbean.getFreeSwapSpaceSize();
         if (min_size > 0) {
             min_size_for_pass = min_size;
         }
 
+        long expected_swap_size = getSwapSizeFromOs();
         long size = mbean.getTotalSwapSpaceSize();
 
-        System.out.println("Total swap space size in bytes: " + size);
+        System.out.println("Total swap space size from OS in bytes: " + expected_swap_size);
+        System.out.println("Total swap space size in MBean bytes: " + size);
+
+        // if swap data from OS chnaged re-read OS and MBean data
+        while (expected_swap_size != getSwapSizeFromOs()) {
```

Comment:
Can we put a limit on this?
OS swap usage could be continually changing, this seems worth a try but we want to escape and give up after some number of tries?
If there is some variation that's making us fail, we could also mask off some number of low bits, or a byte or more even, to do an approximate comparison?  Might want to try that if retrying does not succeed.

Reply:
I doubt that swap might change so often. The failure is reproduced once in several months.
Let's check the logs if the test failed.